### PR TITLE
tests: add test case for precompile implementation 

### PR DESCRIPTION
**Pull request by axic**
Part of #259.

### CODE REVIEW

File: examples/example_precompiles_vm/example_precompiles_vm.cpp [hunk 1]
```diff
@@ -44,7 +44,7 @@ static evmc_result execute_empty(const evmc_message* msg)
 static evmc_result not_implemented()
 {
     auto result = evmc_result{};
-    result.status_code = EVMC_INTERNAL_ERROR;
+    result.status_code = EVMC_REJECTED;
```
Comment (axic) [inline examples/example_precompiles_vm/example_precompiles_vm.cpp]: We have this special case for non-supported code/message. I think it fits better.

Also we have the specialised `EVMC_PRECOMPILE_FAILURE` which we don't use in this example.

File: test/vmtester/tests.cpp [hunk 2]
```diff
@@ -168,3 +168,67 @@ TEST_F(evmc_vm_test, set_tracer)
...
+                                  0,
+                                  0,
+                                  0,
+                                  0,
+                                  0,
+                                  static_cast<uint8_t>(i >> 8),
+                                  static_cast<uint8_t>(i & 0xff)}};
+        evmc_context* context = example_host_create_context();
+        evmc_message msg{
+            EVMC_CREATE,   0, 0, 65536, destination, evmc_address{}, NULL, 0, evmc_uint256be{},
```
Comment (axic) [inline test/vmtester/tests.cpp]: Should change this to `CALL`.

File: test/vmtester/tests.cpp [hunk 3]
```diff
@@ -168,3 +168,67 @@ TEST_F(evmc_vm_test, set_tracer)
...
+            evmc_bytes32{}};
+
+        evmc_result result = vm->execute(vm, context, EVMC_MAX_REVISION, &msg, nullptr, 0);
```
Comment (axic) [inline test/vmtester/tests.cpp]: Should even set `context` to `nullptr` according to the EIP.

**Comment by axic** [comment]
@chfast can I get a review?

File: test/vmtester/tests.cpp (hunk 3 above)
Comment (chfast) [reply test/vmtester/tests.cpp]: Yes.

File: test/vmtester/tests.cpp (hunk 2 above)
Comment (chfast) [reply test/vmtester/tests.cpp]: Yes.

File: test/vmtester/tests.cpp [hunk 4]
```diff
@@ -168,3 +168,67 @@ TEST_F(evmc_vm_test, set_tracer)
...
+
+TEST_F(evmc_vm_test, precompile_test)
+{
+    if (!evmc_vm_has_capability(vm, EVMC_CAPABILITY_PRECOMPILES))
+        return;
+
+    // Iterate every address (as per EIP-1352)
+    for (size_t i = 0; i < 0xffff; i++)
+    {
+        evmc_address destination{{0,
```
Comment (chfast) [inline test/vmtester/tests.cpp]: Looks like giraffe. Maybe write it like:
```
auto destination = evmc_address{};
destination.bytes[18] = static_cast<uint8_t>(i >> 8);
...
```

Comment (axic) [reply test/vmtester/tests.cpp]: Its not a giraffe, but this guy:
image

File: test/vmtester/tests.cpp (hunk 3 above)
Comment (axic) [reply test/vmtester/tests.cpp]: Actually it may be needed to check `callvalue`.

Comment (chfast) [reply test/vmtester/tests.cpp]: Both are available in the message.

Comment (axic) [reply test/vmtester/tests.cpp]: Oh right, that's great.

**Comment by axic** [comment]
I think now this should be in line with https://github.com/ethereum/EIPs/pull/2003

**Pull request merged**